Signal/src/environment/PropertyListPreferences.h

```diff
@@ -61,8 +61,8 @@ extern NSString *const PropertyListPreferencesKeyEnableDebugLog;
 
 - (TSImageQuality)imageUploadQuality;
 
-- (nullable NSString *)lastRanVersion;
-- (NSString *)setAndGetCurrentVersion;
++ (nullable NSString *)lastRanVersion;
++ (NSString *)setAndGetCurrentVersion;
```

Comment:
> Looking at it another way, we would have caught this issue if Environment had no class methods except getCurrent, and all values pulled from that instance rather than the "convenient" class methods (since getCurrent asserts that it's not empty).

The `[Environment preferences]` convenience method uses `[Environment getCurrent]` - so removing it wouldn't help.

For lack of a better option, I'm proposing that we log assert failures for release builds: https://github.com/WhisperSystems/SignalServiceKit/pull/126

> For testability design, I have a slight preference for leaving preferences as a property on the environment instance, rather than relying on more class methods, but I could let that go pretty easily.

I think it would be worse to pre-instantiate a partially working instance of `PropertyListPreferences` before `Environment` was initialized, or any other solution I can think of.